gammapy/background/tests/test_energy_offset_array.py

```diff
@@ -2,24 +2,48 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 from numpy.testing import assert_equal
 import numpy as np
-from ..energy_offset_array import EnergyOffsetArray
 from ...data import DataStore
 from ...datasets import gammapy_extra
 from ...utils.testing import requires_dependency, requires_data
 from ...utils.energy import EnergyBounds
+from ..energy_offset_array import EnergyOffsetArray
+
+
+def make_test_array():
+    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 20, 'TeV')
+    offset = np.linspace(0, 2.5, 10)
+    array = EnergyOffsetArray(ebounds, offset)
+    # put some dummy example data
+    # TODO: this could be made optional or split out into a
+    # separate utility function later on when lookup / interpolation methods are added to the class.
+    array.data[5, 3] = 42
+    return array
 
 
 @requires_dependency('matplotlib')
 @requires_data('gammapy-extra')
-def test_energy_offset_array():
+def test_energy_offset_array_fill():
     dir = str(gammapy_extra.dir) + '/datasets/hess-crab4'
     data_store = DataStore.from_dir(dir)
     ev_list = data_store.load_all('events')
-    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
-    offset = np.linspace(0, 2.5, 100)
-    array = EnergyOffsetArray(ebounds, offset)
+
+    array = make_test_array()
     array.fill_events(ev_list)
+
+    # TODO: add some assert, e.g. counts in some bin with non-zero entries.
+
+
+def test_energy_offset_array_plot():
+    array = make_test_array()
     array.plot_image()
-    array.write("test_write_read_method_energyoffsetarray.fits.gz")
-    array_read=array.read("test_write_read_method_energyoffsetarray.fits.gz")
-    assert_equal(array.data, array_read.data)
+
+
+def test_energy_offset_array_read_write(tmpdir):
+    array = make_test_array()
+
+    filename = str(tmpdir / 'data.fits')
+    array.write(filename)
+    array2 = array.read(filename)
+
+    assert_equal(array.data, array2.data)
+    # TODO: add more asserts for other data members (the axes).
```

Comment:
@JouvinLea - From reading the code it's hard to tell if currently the axes are the same after write / read.

Could you add an assert here that compares `array.energy` and `array.offset` after and before FITS I/O to test if the current methods work?